hivemind/dht/traverse.py

```diff
@@ -149,7 +149,7 @@ def finish_search(query):
             search_finished_event.set()
         if found_callback:
             nearest_neighbors = [peer for _, peer in heapq.nlargest(beam_size, nearest_nodes[query])]
-            pending_tasks.add(asyncio.create_task(found_callback(query, nearest_neighbors, set(visited_nodes))))
+            pending_tasks.add(asyncio.create_task(found_callback(query, nearest_neighbors, set(visited_nodes[query]))))
```

Comment:
(this was a bug, it returned list of queries (visited_nodes.keys() instead of actual nodes)